ads/netletix.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {writeScript, loadScript, validateData} from '../3p/3p';
+import {startsWith} from '../src/string.js';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+/** @type {{nxasync:string}} */
+/** @type {{nxv:string}} */
+/** @type {{nxsite:string}} */
+/** @type {{nxid:string}} */
+/** @type {{nxscript:string}} */
+/** @type {{nxkey:string}} */
+/** @type {{nxunit:string}} */
+/** @type {{nxwidth:string}} */
+/** @type {{nxheight:string}} */
+
+export function netletix(global, data) {
+  /*eslint "google-camelcase/google-camelcase": 0*/
+  global._netletix_amp = {
+    allowed_data: ['nxasync','nxv','nxsite','nxid','nxscript'],
+    mandatory_data: ['nxkey','nxunit','nxwidth','nxheight'],
+    data,
+  };
+
+
+
+  validateData(data,
+      global._netletix_amp.mandatory_data, global._netletix_amp.allowed_data);
+
+  const ls = 'https://call.adadapter.netzathleten-media.de';
+  const rand = Math.round(Math.random() * 100000000);
+  const nxkey = (data.nxkey ? data.nxkey : 'default');
+  const nxunit = (data.nxunit ? data.nxunit : 'default');
+  const nxwidth = (data.nxwidth ? data.nxwidth : 'fluid');
+  const nxheight = (data.nxheight ? data.nxheight : 'fluid');
+  const nxv = (data.nxv ? data.nxv : '0002');
+  const nxsite = (data.nxsite ? data.nxsite : 'none');
+  const url = ls + '/pb/'
```

Comment:
Please use `assertHttpsUrl` and `addParamsToUrl` from the `../src/url` package.

Three other suggestions:
 - Inline the intermediate variable expressions in the params object
 - Use JS-style short-circuiting for default values
- Set module-level constants for your defaults

If you employ all of these, the URL assignment snippet should look like this:

```javascript
const NX_URL_HOST = 'https://call.adadapter.netzathleten-media.de';

const DEFAULT_NX_KEY = 'default';
const DEFAULT_NX_UNIT = 'default';
const DEFAULT_NX_WIDTH = 'fluid';
const DEFAULT_NX_HEIGHT = 'fluid';
const DEFAULT_NX_V = '0002';
const DEFAULT_NX_SITE = 'none';
const DEFAULT_NX_URL = ls + '/pb/'

// ...

export function netletix(global, data) {
  // ...

  const url = assertHttpsUrl(addParamsToUrl(
      NX_URL_HOST + '/pb/' + encodeURIComponent(data.nxkey || DEFAULT_NX_KEY),
      {
        unit: data.nxunit || DEFAULT_NX_UNIT,
        width: data.nxwidth || DEFAULT_NX_WIDTH,
        height:  data.nxheight || DEFAULT_NX_HEIGHT,
        v: data.nxv || DEFAULT_NX_V,
        site: data.nxsite || DEFAULT_NX_SITE,
        ord: Math.round(Math.random() * 100000000),
      }));

  // ...
}
```